src/Scenario/Collector/Step.php

```diff
@@ -0,0 +1,143 @@
+     * @return array
+     * @throws ValidationException
+     */
+    public function collect(array $step): array
+    {
+        $stepName = $step['@name'];
+
+        $arguments = [];
+
+        foreach ($step['arguments']['argument'] ?? [] as $argument) {
+            if ($missedArgs = array_diff(self::$argumentRequiredArgs, array_keys($argument))) {
+                throw new ValidationException(sprintf(
+                    'Argument(s) "%s" are missed from argument in step "%s"',
+                    implode(', ', $missedArgs),
+                    $stepName
+                ));
+            }
+
+            $argumentName = $argument['@name'];
+            $argumentType = $argument['@xsi:type'];
+
+            switch ($argumentType) {
+                case self::XSI_TYPE_ARRAY:
+                    $arguments[] = [
+                        'name' => $argumentName,
+                        'xsi:type' => $argumentType,
+                        'items' => $this->collectItems(
+                            is_array(reset($argument['item'])) ? $argument['item'] : [$argument['item']]
+                        ),
+                    ];
+                    break;
+                case self::XSI_TYPE_OBJECT:
+                case self::XSI_TYPE_STRING:
+                    $arguments[] = [
+                        'name' => $argumentName,
+                        'xsi:type' => $argumentType,
+                        self::NODE_VALUE => $argument[self::NODE_VALUE]
+                    ];
+                    break;
+                default:
+                    throw new ValidationException(sprintf(
+                        'xsi:type "%s" not allowed in argument "%s"',
+                        $argumentType,
+                        $argumentName
+                    ));
+            }
+        }
+
+        $stepData = [
+            'name' => $stepName,
+            'type' => $step['@type'] ?? '',
+            'arguments' => $arguments
+        ];
+
+        if (isset($step['@priority'])) {
+            $stepData['priority'] = intval($step['@priority']);
+        }
+
+        return $stepData;
+    }
+
+    /**
+     * Recursively collect items
+     *
+     * @param array $items
+     * @return array
+     * @throws ValidationException
+     */
+    private function collectItems(array $items): array
+    {
+        $newItems = [];
+
+        foreach ($items as $item) {
+            $itemName = $item['@name'] ?? '';
+
+            if ($missedArgs = array_diff(self::$itemRequiredArgs, array_keys($item))) {
+                throw new ValidationException(sprintf(
+                    'Argument(s) "%s" are missed from item "%s"',
+                    implode(', ', $missedArgs),
+                    $itemName
+                ));
+            }
+
+            $newItem = [
+                'name' => $itemName,
+                'xsi:type' => $item['@xsi:type'],
+                'priority' => $item['@priority'] ?? self::DEFAULT_PRIORITY,
```

Comment:
Priority should be required

Reply:
Priority is required for steps, but not required for items as not all item tags need priority,
for example:
```
<item name="500" xsi:type="array">
                    <item name="database-configuration" xsi:type="object">Magento\MagentoCloud\Config\Validator\Deploy\DatabaseConfiguration</item>
                    <item name="search-configuration" xsi:type="object">Magento\MagentoCloud\Config\Validator\Deploy\SearchConfiguration</item>
                </item>
```
in this case <item name="500" xsi:type="array"> doesn't need a priority